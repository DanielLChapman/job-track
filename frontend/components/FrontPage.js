import React from 'react';

function FrontPage(props) {
    return (
        <div>
            Hi!
        </div>
    );
}

export default FrontPage;