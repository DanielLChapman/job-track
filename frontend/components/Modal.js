import React from "react";
import { CSSTransition } from "react-transition-group";

function Modal(props) {
    let t = props.classes;
    return (
        <>
            <div
                className="blur-layer"
                onClick={() => {
                    props.closeFunc(false);
                }}
            ></div>
            <div className={` modal-window ${t} `}>
                <button
                    type="button"
                    className="close-modal"
                    onClick={() => {props.closeFunc(false);}}
                >
                    X
                </button>
                {props.children}
            </div>
        </>
    );
}

export default Modal;
