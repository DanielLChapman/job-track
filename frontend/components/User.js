import { useQuery } from "@apollo/client";
import gql from "graphql-tag";


export const CURRENT_USER_QUERY = gql`
    query {
        authenticatedItem {
            ... on User {
                id 
                name
                jobs {
                    name
                    publishedAt
                    applicationDate
                    status
                    salaryExpectation
                    interviews {
                        date 
                        type 
                    }
                    notes
                    id
                }
            }
        }
    }
`;

export function useUser() {
    const {data, error, loading} = useQuery(CURRENT_USER_QUERY);
    return data?.authenticatedItem;
}

