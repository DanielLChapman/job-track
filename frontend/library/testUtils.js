import casual from 'casual';

casual.seed(231313);

const fakeUser = (overrides) => ({
    __typename: 'User',
    id: '4234',
    name: casual.name,
    jobs: [],
    email: casual.email,
    ...overrides,
  });

  const fakeJob = (overrides) => ({
    __typename: 'User',
    id: '4234',
    name: casual.name,
    publishedAt: casual.date('YYYY-MM-DD'),
    applicationDate: casual.date('YYYY-MM-DD'),
    status: "waiting",
    author: fakeUser().id,
    salaryExpectation: 0,
    notes: casual.sentence,
    ...overrides,
  });

  export {
    fakeUser,
    fakeJob
  }
