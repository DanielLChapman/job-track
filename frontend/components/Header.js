import React, { useState } from "react";
import PropTypes from "prop-types";
import CreateJob from "./Jobs/CreateJob";
import SignOut from "./Signout";
import SignIn from "./SignIn";
import SignUp from "./SignUp";
import Modal from "./Modal";
import { CSSTransition } from "react-transition-group";

Header.propTypes = {
    user: PropTypes.object,
};

function Header({ user }) {
    const [modal, setModal] = useState(false);
    const [modalCreateJob, setModalCreateJob] = useState(false);
    const [inUp, setInUp] = useState("In");

    return (
        <>
            <h1>Job Track</h1>
            {user && (
                <div className="header-button-class">
                    <button
                        type="button"
                        className="sign-in-button"
                        onClick={() => {
                            setModalCreateJob(!modalCreateJob);
                        }}
                    >
                        Create Job
                    </button>

                    <SignOut />
                    {modalCreateJob && (
                        <Modal closeFunc={setModalCreateJob}>
                            <CreateJob user={user} />
                        </Modal>
                    )}
                </div>
            )}
            {!user && (
                <div className="header-button-class">
                    <button
                        type="button"
                        className="sign-in-button"
                        onClick={() => {
                            setModal(!modal);
                        }}
                    >
                        Sign In / Up
                    </button>
                </div>
            )}
            {!user && (
                <CSSTransition
                    in={modal}
                    timeout={200}
                    classNames="modal-transition"
                    unmountOnExit
                    onExited={() => setModal(false)}
                >
                    
                    <Modal closeFunc={setModal} >
                        <div className="header">
                            {inUp === "In" && (
                                <button
                                    type="button"
                                    className="sign-in-up-selector-button"
                                    onClick={() => {
                                        setInUp("Up");
                                    }}
                                >
                                    Sign Up
                                </button>
                            )}
                            {inUp === "Up" && (
                                <button
                                    type="button"
                                    className="sign-in-up-selector-button"
                                    onClick={() => {
                                        setInUp("In");
                                    }}
                                >
                                    Sign In
                                </button>
                            )}
                        </div>
                        {inUp === "In" && (
                            <SignIn closeFunc={setModal} closeValue={!modal} />
                        )}
                        {inUp === "Up" && <SignUp />}
                    </Modal>
                </CSSTransition>
            )}
        </>
    );
}

export default Header;
