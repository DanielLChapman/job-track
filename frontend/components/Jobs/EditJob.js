import React from "react";
import PropTypes from "prop-types";
import gql from "graphql-tag";
import { useMutation } from "@apollo/client";
import useForm from "../../library/useForm";
import { CURRENT_USER_QUERY } from "../User";

EditJob.propTypes = {
    job: PropTypes.object.isRequired,
    closeForm: PropTypes.func
};

export const EDIT_JOB_MUTATION = gql`
    mutation EDIT_JOB_MUTATION(
        $name: String!
        $status: String
        $id: ID!
        $salaryExpectation: Int
        $notes: String
    ) {
        updateJob(
            where: { id: $id }
            data: {
                name: $name
                status: $status
                salaryExpectation: $salaryExpectation
                notes: $notes
            }
        ) {
            id
        }
    }
`;

function EditJob(props) {

    let job = props.job;

    const { inputs, handleChange } = useForm({
        name: job.name,
        status: job.status || "waiting",
        id: job.id,
        salaryExpectation: job.salaryExpectation,
        notes: job.notes,
    });

    const [updateJob, { data, error, loading }] = useMutation(
        EDIT_JOB_MUTATION,
        {
            variables: {
                ...inputs,
            },
            refetchQueries: [{ query: CURRENT_USER_QUERY }],
        }
    );

    const handleSubmit = async (e) => {
        e.preventDefault();
        const res = await updateJob();
        if (res.data.updateJob.id) {
            props.closeForm(false);
        }
    };

    return (
        <>
            {error && <span>Error, Please Try Again</span>}
            <form method="POST" onSubmit={handleSubmit}>
                <h2>Edit {props.job.name}</h2>
                <fieldset disabled={loading} aria-busy={loading}>
                    <label htmlFor="name">
                        Name
                        <input
                            type="text"
                            name="name"
                            placeholder="Name"
                            value={inputs.name}
                            onChange={handleChange}
                        />
                    </label>
                    <label htmlFor="status">
                        Status
                        <select
                            name="status"
                            value={inputs.status}
                            onChange={handleChange}
                        >
                            <option value="waiting">Waiting</option>
                            <option value="rejected">Rejected</option>
                            <option value="accepted">Accepted</option>
                        </select>
                    </label>
                    <label htmlFor="salaryExpectation">
                        Salary Expectation
                        <input
                            type="number"
                            name="salaryExpectation"
                            placeholder="0"
                            value={inputs.salaryExpectation}
                            onChange={handleChange}
                        />
                    </label>
                    <label htmlFor="notes">
                        Notes
                        <textarea
                            value={inputs.notes}
                            onChange={handleChange}
                            name="notes"
                        />
                    </label>

                    <button type="submit">Edit Job</button>
                </fieldset>
            </form>
        </>
    );
}

export default EditJob;
