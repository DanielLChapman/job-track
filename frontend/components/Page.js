import PropTypes from "prop-types";
//import Header from './Header';
import Head from "next/head";
import { createGlobalStyle } from "styled-components";
import { useUser } from "./User";

const GlobalStylesSignedIn = createGlobalStyle`

  html, *, *:before, *:after {
    box-sizing: inherit;
    margin: 0;
    padding: 0;
  }
  body {
    font-family: --apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height:2;
    height: auto;
    background: rgb(2, 0, 36);
    background: linear-gradient(
        0deg,
        rgba(0, 212, 255, 0.5) 0%,
        rgba(0, 212, 255, 1) 100%
    );
  }
  a {
    text-decoration: none;
    color: var(--black);
  }
  a:hover {
    text-decoration: underline;
  }
  button {
    font-family: --apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  }
`;

const GlobalStylesSignedOut = createGlobalStyle`

  html, *, *:before, *:after {
    box-sizing: inherit;
    margin: 0;
    padding: 0;
  }
  body {
    font-family: --apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height:2;
    background:url('./background-signed-out.jpg') no-repeat center center;
    background-size: cover;
    height: auto;
  }
  a {
    text-decoration: none;
    color: var(--black);
  }
  a:hover {
    text-decoration: underline;
  }
  button {
    font-family: --apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  }
`;

export default function Page(props) {
    const user = useUser();
    
    return (
        <div>
            <Head>
                <meta
                    name="viewport"
                    content="initial-scale=1.0, width=device-width"
                />
            </Head>
            {!user && <GlobalStylesSignedOut />}
            {user && <GlobalStylesSignedIn />}

            {/*<Header />*/}
            <main>{props.children}</main>
            <link rel="stylesheet" href="main.css" />
            
        </div>
    );
}

Page.propTypes = {
    children: PropTypes.any,
};
