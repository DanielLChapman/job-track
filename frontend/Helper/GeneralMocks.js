
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser } from '../library/testUtils';

const fakeUserObj = fakeUser();
const notSignedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: null } },
    },
  ];
  

  const signedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: fakeUser() } },
    },
  ];

  export {
    notSignedInMocks,
    signedInMocks,
}