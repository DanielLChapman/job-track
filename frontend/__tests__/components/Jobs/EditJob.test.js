import {act, screen, render} from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import {signedInMocks, notSignedInMocks} from '../../../Helper/GeneralMocks';
import wait from 'waait';
import { fakeJob, fakeUser } from '../../../library/testUtils';
import EditJob, {EDIT_JOB_MUTATION} from '../../../components/Jobs/EditJob';
import { CURRENT_USER_QUERY } from '../../../components/User';
import userEvent from '@testing-library/user-event';


const job = fakeJob();
const user = fakeUser();
const signedInMocks3 = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: job.author } },
    },
  ];


const mocks = [
    {
        request: {
            query: EDIT_JOB_MUTATION,
            variables: {
                name: job.name,
                status: job.status,
                id: job.id,
                salaryExpectation: job.salaryExpectation,
                notes: job.notes,
            },
        },
        result: {
            data: {
                updateJob: {
                    id: job.id,
                    __typename: 'Job',
                }
            }
        }
    },
    {
        request: { query: CURRENT_USER_QUERY },
        result: { data: { authenticatedItem: job.author } },
    },
]

describe('it renders a job edit form and submits successfully', () => {
    it ('does not load unless signed in', async() => {
        const {container, debug} = render(
            <MockedProvider>
                <EditJob job={job} id={job.id} closeForm={() => {}}/>
            </MockedProvider>
        )

        await screen.findByText(
            'Please Log In'
        );

        expect(container).toMatchSnapshot();
    })

    it ('does not load unless signed in and user is the same user as the job', async() => {
        const {container, debug} = render(
            <MockedProvider mocks={signedInMocks}>
                <EditJob job={job} id={job.id} closeForm={() => {}}/>
            </MockedProvider>
        )

        await screen.findByText(
            "You Shouldn't Be Here"
        );

        expect(container).toMatchSnapshot();
    })

    it('calls the mutation properly', async () => {

        const {container, debug} = render(
            <MockedProvider mocks={mocks}>
                <EditJob job={job} id={job.id} closeForm={() => {}}/>
            </MockedProvider>
        )

        await screen.findByText(
            `Edit ${job.name}`
        );

        // Type into the boxes
        //await userEvent.type(screen.getByPlaceholderText(/name/i), "1");
        //await userEvent.type(screen.getByTestId('notes'), "2");
      // Click the submit
        await userEvent.click(screen.getByText('Edit Job'));
        await act(async () => {
          await wait(0);
        })
        
        const element = await screen.getByTestId('success-message');
        expect(element);
        /*await screen.findByText(
          `Signed Up With`
        );*/
      });

    /*
    it ('does load correctly unless signed in', async() => {
        const {container, debug} = render(
            <MockedProvider mocks={signedInMocks}>
                <CreateJob user={fakeUser()} />
            </MockedProvider>
        )

        await screen.findByText(
            'Create a new Job'
        );

        expect(container).toMatchSnapshot();
    })*/
})