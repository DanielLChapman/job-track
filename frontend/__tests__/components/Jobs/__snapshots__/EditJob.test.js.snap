// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a job edit form and submits successfully does not load unless signed in 1`] = `
<div>
  <span>
    Please Log In
  </span>
</div>
`;

exports[`it renders a job edit form and submits successfully does not load unless signed in and user is the same user as the job 1`] = `
<div>
  <span>
    You Shouldn't Be Here
  </span>
</div>
`;
